protostar/commands/init/project_creator/new_project_creator.py

```diff
@@ -52,12 +66,85 @@ def _gather_input(self) -> "NewProjectCreator.NewProjectConfig":
 
     def _create_project(
         self,
-        user_input: "NewProjectCreator.NewProjectConfig",
+        project_config: "NewProjectCreator.NewProjectConfig",
         cairo_version: CairoVersion,
     ) -> None:
         output_dir_path = self._output_dir_path
-        project_root_path = output_dir_path / user_input.project_dirname
-        self.copy_template(cairo_version, project_root_path)
-        self.save_protostar_toml(
-            project_root_path=project_root_path, cairo_version=cairo_version
+        project_root_path = output_dir_path / project_config.project_dirname
+        project_name = project_config.project_dirname
+
+        self.copy_and_replace_template(
+            cairo_version=cairo_version,
+            project_root_path=project_root_path,
+            project_name=project_name,
+        )
+        self._save_protostar_toml(
+            project_root_path=project_root_path,
+            configuration_model=self._make_project_config(cairo_version=cairo_version),
+        )
+
+    def copy_and_replace_template(
+        self, cairo_version: CairoVersion, project_root_path: Path, project_name: str
+    ):
+        template_path = self.script_root / "templates" / cairo_version.value
+
+        with tempfile.TemporaryDirectory() as temp_dir:
+            temp_dir = Path(temp_dir)
+            new_template_path = Path(temp_dir) / "template"
+
+            shutil.copytree(template_path, new_template_path)
+            self._insert_project_name_to_template(project_name, new_template_path)
+
+            try:
+                shutil.copytree(new_template_path, project_root_path)
+            except FileExistsError as ex_file_exists:
+                raise ProtostarException(
+                    f"Folder or file named {project_root_path.name} already exists. Choose different project name."
+                ) from ex_file_exists
+
+    def _make_project_config(
+        self, cairo_version: CairoVersion
+    ) -> ConfigurationFileV2Model:
+        if cairo_version == CairoVersion.cairo0:
+            return ConfigurationFileV2Model(
+                protostar_version=str(self._protostar_version),
+                contract_name_to_path_strs={"main": ["src/main.cairo"]},
+                project_config={
+                    "lib-path": "lib",
+                },
+                command_name_to_config={},
+                profile_name_to_project_config={},
+                profile_name_to_commands_config={},
+            )
+
+        return ConfigurationFileV2Model(
+            protostar_version=str(self._protostar_version),
+            contract_name_to_path_strs={"hello_starknet": ["src"]},
```

Comment:
This doesn't point to a file, why?

Reply:
Good question, not sure at this point. This definitely need to point to a specific contract in case multiple are present in `contracts` directory.

Reply:
For this project structure to work `hello_starknet.cairo` references code in `business_logic/` dir, compilation must be in the context of whole `src`. Which is problematic because it will not work if there are multiple contracts in `contracts/` dir due to cairo1 compiler limitation.